### DISCUSSION TITLE
PICO W fails socket.getaddrinfo

### DISCUSSION THREAD

**Discussion by miltmobley** [micropython/micropython#10033] [Using MicroPython] [answered]
I am following the steps in RPi's document 'Connecting to the internet with ... pico w'. I can get down to the line:
    ai = socket.getaddrinfo('google.com', 80)

After a few seconds I get an exception message with cause 'OSError: -2'

Due to the delay, I suspect this is a timeout. If so, what can I do about it?

**Comment by peterhinch** [comment, marked as answer]
Well, this line works on my Pico W, returning instantly, so it seems your DNS query is failing. I would suspect a network problem: can the Pico access the internet? Is a DNS server configured?

**Reply by miltmobley** [reply]
I am embarrassed! I misspelled the SSID name, now everything seems to work.
